gslb/cache/controller_obj_cache.go

```diff
@@ -613,18 +644,14 @@ func GetDetailsFromAviGSLB(gslbSvcMap map[string]interface{}) (uint32, []GSMembe
 	}
 
 	var poolAlgorithmSettings *gslbalphav1.PoolAlgorithmSettings
+
 	for _, val := range groups {
 		group, ok := val.(map[string]interface{})
 		if !ok {
 			gslbutils.Warnf("couldn't parse group: %v", val)
 			continue
 		}
-		gsGroup, ok := val.(models.GslbPool)
-		if !ok {
-			gslbutils.Warnf("couldn't parse group to GslbPool: %v", gsGroup)
-			continue
-		}
-		poolAlgorithmSettings = ParsePoolAlgorithmSettings(gsGroup)
+		poolAlgorithmSettings = ParsePoolAlgorithmSettingsFromPoolRaw(group)
```

Comment:
check poolAlgorithmSettings is nil / return error to handle the error from function.

Reply:
Actually, its taken care of in the checksum calculation which is the only place where this variable is getting used.

Reply:
Yes. Searched that it was going to the checksum function,  later did not check the usage.